src/shared/utilities/timeoutUtils.ts

```diff
@@ -118,3 +123,41 @@ export async function waitUntil<T>(
         await new Promise(r => setTimeout(r, opt.interval))
     }
 }
+
+/**
+ * Utility function to wrap a Timeout token around a Promise.
+ *
+ * @param promise The promise to use a Timeout with
+ * @param timeout A Timeout token that will race against the promise
+ * @param opt.noUndefined Prevents the promise from being resolved undefined (default: false)
```

Comment:
If we avoid the double negative, is it easier to explain (and understand)? 

```suggestion
 * @param opt.defined Ensures the promise will resolve non-undefined (default: true)
```